internal/resources/resource_volume.go

```diff
@@ -288,6 +320,8 @@ func resourceMetalVolumeRead(d *schema.ResourceData, meta interface{}) (err erro
 		}
 	}
 
+	_ = d.Set(vStoragePoolID, volume.StoragePoolID)
+
 	return nil
```

Comment:
Do we really want to throw away the return from `d.Set()`?
```suggestion
	err := d.Set(vStoragePoolID, volume.StoragePoolID)

	return err
```